### DISCUSSION TITLE
what does it mean to collect rewards?

### DISCUSSION THREAD

**Discussion by paxelpixel** [w3f/staking-rewards-collector#26] [Q&A]
from the tutorial.it sounds like this is just a calculator? or does it actually collect rewards like what you can do from the polkadot js app?

**Comment by jonasW3F** [comment]
> from the tutorial.it sounds like this is just a calculator? or does it actually collect rewards like what you can do from the polkadot js app?

It just collects the staking rewards information and does some cross-referencing with daily prices and produces an output file. It does not interact directly with the chain and your rewards since that would require access to your account, which is not a thing we would want to do.

Essentially it can help you with bookkeeping of your rewards.

**Comment by paxelpixel** [comment]
ah that is good to know! this is something I can add to polkaview.network! very nice

**Comment by paxelpixel** [comment]
![Screen Shot 2021-06-05 at 7 35 51 PM](https://user-images.githubusercontent.com/13143458/120910670-81e88e00-c635-11eb-9f31-2797d0a9b541.png)
![Screen Shot 2021-06-05 at 7 36 32 PM](https://user-images.githubusercontent.com/13143458/120910672-8319bb00-c635-11eb-8f0f-77cc257cdc85.png)

WIP so excited

**Comment by paxelpixel** [comment]
![Screen Shot 2021-06-05 at 7 39 46 PM](https://user-images.githubusercontent.com/13143458/120910705-d55adc00-c635-11eb-92d6-7b43c5362c16.png)
mobile view will be a scrollable table

**Comment by jonasW3F** [comment]
Looks cool, are you making a UI for the collector? Are you aware that there is currently an RFP for exactly that? You might want to look into that here: https://github.com/w3f/General-Grants-Program/pull/421. cc: @mmagician

**Reply by paxelpixel** [reply]
yes, thank you! I learned it from bill laboon. but before I apply for proposal I'm curious what you think so far when you get a chance

here is a calculation link with 2 addresses 1 kusama and 1 polkadot.  
https://www.stakingrewards.network/rewards?addresses=%5B%7B%22name%22%3A%22%22%2C%22startBalance%22%3A%2210000%22%2C%22address%22%3A%22G1rrUNQSk7CjjEmLSGcpNu72tVtyzbWdUvgmSer9eBitXWf%22%2C%22error%22%3Anull%2C%22network%22%3A%22kusama%22%7D%2C%7B%22name%22%3A%22%22%2C%22startBalance%22%3A%2220000%22%2C%22address%22%3A%2215fTw39Ju2jJiHeGe1fJ5DtgugUauy9tr2HZuiRNFwqnGQ1Q%22%2C%22error%22%3Anull%2C%22network%22%3A%22polkadot%22%7D%5D&endDate=31&endMonth=1&endYear=2021&selectedCurrency=CHF&startDate=1&startMonth=1&startYear=2021

you can add custom address here
https://www.stakingrewards.network

thank you very much. I look forward to what you think! things to note is that I converted backend specific npm packages to browser friendly one. for e.g curl -> fetch 

I'm currently working on daily view and exports.

**Comment by paxelpixel** [comment]
I noticed the kusama are not going beyond the jan 3rd 2021 even if I set the userInput later. am I missing something? 

https://www.stakingrewards.network/rewards?addresses=%5B%7B%22name%22%3A%22%22%2C%22startBalance%22%3A%22%22%2C%22address%22%3A%22G1rrUNQSk7CjjEmLSGcpNu72tVtyzbWdUvgmSer9eBitXWf%22%2C%22error%22%3Anull%2C%22network%22%3A%22kusama%22%7D%2C%7B%22name%22%3A%22%22%2C%22startBalance%22%3A%22%22%2C%22address%22%3A%2215fTw39Ju2jJiHeGe1fJ5DtgugUauy9tr2HZuiRNFwqnGQ1Q%22%2C%22error%22%3Anull%2C%22network%22%3A%22polkadot%22%7D%5D&endDate=3&endMonth=6&endYear=2021&selectedCurrency=USD&startDate=1&startMonth=12&startYear=2020

**Reply by paxelpixel** [reply]
nvm looks like an issue with the API. it stops sending kusama rewards data after that date. when I look at this tool
https://www.cryptolab.network/tools/dotSR its sending just find for 
G1rrUNQSk7CjjEmLSGcpNu72tVtyzbWdUvgmSer9eBitXWf

**Reply by paxelpixel** [reply]
I'm running an archive node for https://polkaview.network/dot/identities what do you think I just query the blockchain for a certain era. its gonna take longer but it will be fresh on-chain data. and its gonna run client side

**Reply by paxelpixel** [reply]
is it practical?

**Reply by jonasW3F** [reply]
My tool has no issues going further back than 3rd jan. I am not sure if that is practical. I think it has quite a long runtime querying the blockchain itself. This could be a cool extension though, like users who do not want to rely on subscan API can specify the option to get the data from the chain (and are fine with a very long runtime).
